.github/workflows/selfdrive_tests.yaml

```diff
@@ -19,7 +19,7 @@ env:
   DOCKER_LOGIN: docker login ghcr.io -u ${{ github.actor }} -p ${{ secrets.GITHUB_TOKEN }}
   BUILD: selfdrive/test/docker_build.sh base
 
-  RUN: docker run --shm-size 1G -v $PWD:/tmp/openpilot -w /tmp/openpilot -e CI=1 -e PRE_COMMIT_HOME=/tmp/pre-commit -e PYTHONWARNINGS=error -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/pre-commit:/tmp/pre-commit -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
```

Comment:
what happened to the cache? if it's causing problems, we need to fix it

Reply:
Building with the cache resulted in those types of cythons errors: https://github.com/commaai/openpilot/actions/runs/9670899274/job/26680488268#step:6:204

I'm trying to find out why, but very annoying since only happening in github actions

Reply:
@deanlee maybe you want to look into this?

Reply:
looks like a NumPy version issue.
the following includes has been added to the generated xxx_pyx.cpp file
```
    /* Using NumPy API declarations from "numpy/__init__.cython-30.pxd" */
    
#include "numpy/arrayobject.h"
#include "numpy/ndarrayobject.h"
#include "numpy/ndarraytypes.h"
#include "numpy/arrayscalars.h"
#include "numpy/ufuncobject.h"
```
Those undeclared identifiers are functions that are new in numpy 2.0.

> msgq_repo/msgq/visionipc/visionipc_pyx.cpp:22567:13: error: use of undeclared identifier 'PyDataType_ELSIZE'
>  22567 |   __pyx_r = PyDataType_ELSIZE(__pyx_v_self);
>        |             ^
> msgq_repo/msgq/visionipc/visionipc_pyx.cpp:22601:13: error: use of undeclared identifier 'PyDataType_ALIGNMENT'
>  22601 |   __pyx_r = PyDataType_ALIGNMENT(__pyx_v_self);
>        |             ^
> msgq_repo/msgq/visionipc/visionipc_pyx.cpp:22639:15: error: use of undeclared identifier 'PyDataType_FIELDS'
>  22639 |   __pyx_t_1 = PyDataType_FIELDS(__pyx_v_self);

       |               ^

Reply:
I can reproduce this with building openpilot with numpy 2.0.0, then installing numpy 1.26.4 and rebuilding openpilot. 

The first commit used numpy 2.0.0, but I changed it back to 1.26.4 in the next one. Seems like it tries to use scons cache that was somehow generated during that first commit.